### PR TITLE
feat: less dependencies limits

### PR DESCRIPTION
**Pull request by ca11ab1e**
### What I did

In order to ease future updates, I reduced all dependencies limits.

Most changes are same on https://github.com/ApeWorX/ape-starknet/pull/22 where you can find some comments.

**Note**: this is a PR that needs other PRs to be merged on other projects in order to fix everything. Let's start with that `ape-cairo`, and move forward on other modules when a new release of `ape-cairo` will be out.

### Checklist

- [ ] Passes all linting checks (pre-commit and CI jobs)
- [ ] New test cases have been added and are passing
- [ ] Documentation has been updated
- [x] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)
- [ ] #17

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -4,24 +4,24 @@
...
         "pytest-cov",  # Coverage analyzer plugin
-        "hypothesis>=6.2.0,<7.0",  # Strategy-based fuzzer
+        "hypothesis",  # Strategy-based fuzzer
     ],
     "lint": [
-        "black>=22.3.0,<23.0",  # auto-formatter and linter
-        "mypy>=0.960,<1.0",  # Static type analyzer
-        "flake8>=4.0.1,<5.0",  # Style linter
-        "isort>=5.10.1,<6.0",  # Import sorting linter
+        "black",  # auto-formatter and linter
```
Comment (unparalleled-js) [inline setup.py:13]: These minimally should be the same as what its in `pre-commit.yaml`.
The upper limits can go though.

File: setup.py [hunk 2]
```diff
@@ -4,24 +4,24 @@
...
+        "flake8",  # Style linter
+        "isort",  # Import sorting linter
     ],
     "release": [  # `release` GitHub Action job uses this
         "setuptools",  # Installation tool
         "wheel",  # Packaging tool
         "twine",  # Package upload tool
     ],
     "dev": [
-        "commitizen>=2.19,<2.20",  # Manage commits and publishing releases
```
Comment (unparalleled-js) [inline setup.py:24]: The version pin exists for a reason... It helps lessen some madness that sometimes occurs when install `[dev]` dependencies... It is hard to explain. I'd advise to leave it for now.

File: setup.py [hunk 3]
```diff
@@ -53,13 +53,9 @@
...
     include_package_data=True,
     install_requires=[
-        "ecdsa",  # Not directly used, but part of install instructions for cairo-lang
-        "fastecdsa",  # Not directly used, but part of install instructions for cairo-lang
-        "sympy",  # Not directly used, but part of install instructions for cairo-lang
-        "cairo-lang",
-        "starknet.py>=0.2.3a0,<0.3.0",
-        "eth-ape>=0.2.7,<0.3.0",
-        "ethpm-types",
+        "cairo-lang>=0.9.0",
```
Comment (unparalleled-js) [inline setup.py:56]: These changes are needed in the other branch so maybe we should just merge these branches

File: setup.py [hunk 4]
```diff
@@ -53,13 +53,9 @@
...
+        "starknet.py>=0.3.1a0",
+        "eth-ape>=0.2.8",
```
Comment (unparalleled-js) [inline setup.py:58]: We plan our breaking changes on the major releases, so the plugins must be constrained to  `<0.3`.
We also pair the versioning of the plugins to the ape core version.

**Comment by ca11ab1e** [comment]
Thanks for the returns, I'll update the PR when #18 will get merged.

**Comment by ca11ab1e** [comment]
Closing that one finally. #17 will be enough to move forward.

**Pull request closed without merging**